asset_dashboard/static/js/PortfolioPlanner.js

```diff
@@ -332,25 +405,40 @@ class PortfolioPlanner extends React.Component {
     }))
   }
 
-  createRegionName(regions) {
-    // returns a CSV string of names for the different regions
-    const names = regions.map(({ name }) => {
-      return name
-    }).join(',')
-
-    return names
-  }
-
   getDate() {
     const date = new Date(new Date().toString().split('GMT')[0]+' UTC').toISOString().split('T')[0]
     return date
   }
+  
+  changeSection(e) {
+    const newSection = e.target.value
+    this.setState(prevState => {
+      return {
+        ...prevState,
+        selectedSection: newSection
+      }
+    })
+  }
 
   render() {
-    // This filters on every re-render, so that this.state.remainingProjects can be the source of truth.
-    // Could also chain filtering here for other things (like "department" from the wireframe).
     const filteredRows = this.state.remainingProjects && this.state.remainingProjects.filter(project => {
       return project.description.toLowerCase().includes(this.state.filterText.toLowerCase())
+    }).filter(project => {
+      if (this.state.selectedSection !== '') {
+        return project.section.toLowerCase().includes(this.state.selectedSection.toLowerCase())
+      } else {
+        return project
+      }
+    })
+    
+    const filteredPortfolio = this.state.portfolio.projects && this.state.portfolio.projects.filter(project => {
```

Comment:
what about optional chaining here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
optional chaining would [return undefined if we used it on an array method](https://stackoverflow.com/a/60165273). i'm afraid that might blow up the table since the table needs an iterable, but i'll try. it's possible that we don't need any truthy check whatsoever, since this code is an artifact from the CIP prototype.

Reply:
i see. wouldn't your code return an undefined right now if projects was not defined?

Reply:
good point. i'll rework this.